wordpress-mobile/WordPress-Android
# P2 people invite links analytics

Part of #13456

This PR adds tracking to the invite links feature.

The following are the added tracks:

**Loading invite links status (opening the Invite People screen)**
```
🔵 Tracked: invite_links_get_status, Properties: {"invite_links_action_result":"succeeded","invite_links_action_has_links":true,"blog_id":001"is_jetpack":false,"site_type":"p2"}
```

**Generating links**
```
🔵 Tracked: invite_links_generate, Properties: {"invite_links_action_result":"succeeded","invite_links_action_has_links":true,"blog_id":001,"is_jetpack":false,"site_type":"p2"}
```

**Sharing a link**
```
🔵 Tracked: invite_links_share, Properties: {"invite_links_action_result":"succeeded","invite_links_shared_role":"administrator","blog_id":001,"is_jetpack":false,"site_type":"p2"}
````

**Disabling links**
```
🔵 Tracked: invite_links_disable, Properties: {"invite_links_action_result":"succeeded","invite_links_action_has_links":false,"blog_id":001,"is_jetpack":false,"site_type":"p2"}
```

Note that it is possible to get `invite_links_action_result` with error state like for example when in flight mode and attempting an invite link action

```
🔵 Tracked: invite_links_generate, Properties: {"invite_links_action_result":"error","invite_links_action_error_message":"no_network","blog_id":001,"is_jetpack":false,"site_type":"p2"}
```

## To test

- Be sure you have tracks enabled in the app settings
- Place in logcat a filter like `Tracked: `
- Check you get the expected tracks as reported above when doing the relevant actions
- Check you get errors reported in the tracks when in flight mode or forcing other error conditions using the debugger
- Smoke test the feature

PR submission checklist:
